common/openapi.yml

```diff
@@ -52,7 +52,10 @@ paths:
           $ref: '#/components/responses/ServerError'
     post:
       tags: [ datasets ]
-      description: Create a new dataset
+      description: |
+        ⚠️ Do not add sensitive data. All catalog entries are publicly accessible to all Terra users ⚠️
```

Comment:
Maybe this should be a note on the `metadata` property in `CreateDatasetRequest` instead of a general warning on the API, since the warning applies to that property specifically.

Reply:
I think we want to make it as visible as possible, so we should add it to the main page and on the `CreateDatasetRequest` object too.